subgraphs/burrow/src/helpers/protocol.ts

```diff
@@ -1,119 +1,161 @@
+    protocol.type = ProtocolType.LENDING;
+    protocol.lendingType = LendingType.POOLED;
+    protocol.riskType = RiskType.GLOBAL;
+    protocol.totalValueLockedUSD = BD_ZERO;
+    protocol.cumulativeUniqueUsers = 0;
+    protocol.cumulativeSupplySideRevenueUSD = BD_ZERO;
+    protocol.cumulativeProtocolSideRevenueUSD = BD_ZERO;
+    protocol.cumulativeTotalRevenueUSD = BD_ZERO;
+    protocol.totalPoolCount = 0;
+    protocol._marketIds = new Array<string>();
+    protocol.save();
+  }
+  return protocol;
 }
 
 export function getOrCreateUsageMetricsDailySnapshot(
-	receipt: near.ReceiptWithOutcome
+  receipt: near.ReceiptWithOutcome
 ): UsageMetricsDailySnapshot {
-	const timestamp = receipt.block.header.timestampNanosec / 86400000000000;
-	const id = timestamp.toString();
-	let usageMetricsDailySnapshot = UsageMetricsDailySnapshot.load(id);
-	const protocol = getOrCreateProtocol()
-	if (!usageMetricsDailySnapshot) {
-		usageMetricsDailySnapshot = new UsageMetricsDailySnapshot(id);
-		usageMetricsDailySnapshot.dailyActiveUsers = 0;
-		usageMetricsDailySnapshot.cumulativeUniqueUsers = protocol.cumulativeUniqueUsers;
-		usageMetricsDailySnapshot.cumulativeUniqueDepositors = protocol.cumulativeUniqueDepositors;
-		usageMetricsDailySnapshot.cumulativeUniqueBorrowers = protocol.cumulativeUniqueBorrowers;
-		usageMetricsDailySnapshot.dailyActiveLiquidators = 0;
-		usageMetricsDailySnapshot.cumulativeUniqueLiquidators = protocol.cumulativeUniqueLiquidators;
-		usageMetricsDailySnapshot.dailyActiveLiquidatees = 0;
-		usageMetricsDailySnapshot.cumulativeUniqueLiquidatees = protocol.cumulativeUniqueLiquidatees;
-		usageMetricsDailySnapshot.dailyTransactionCount = 0;
-		usageMetricsDailySnapshot.dailyDepositCount = 0;
-		usageMetricsDailySnapshot.dailyWithdrawCount = 0;
-		usageMetricsDailySnapshot.dailyBorrowCount = 0;
-		usageMetricsDailySnapshot.dailyRepayCount = 0;
-		usageMetricsDailySnapshot.dailyLiquidateCount = 0;
-		usageMetricsDailySnapshot.totalPoolCount = protocol._marketIds.length;
-		usageMetricsDailySnapshot.blockNumber = BigInt.fromU64(receipt.block.header.height);
-		usageMetricsDailySnapshot.timestamp = BigInt.fromU64(receipt.block.header.timestampNanosec / 1000000000);
-		usageMetricsDailySnapshot.save();
-	}
-	return usageMetricsDailySnapshot as UsageMetricsDailySnapshot;
+  const timestamp = NANOS_TO_DAY(receipt.block.header.timestampNanosec);
+  const id = timestamp.toString();
+  let usageMetricsDailySnapshot = UsageMetricsDailySnapshot.load(id);
+  const protocol = getOrCreateProtocol();
+  if (!usageMetricsDailySnapshot) {
+    usageMetricsDailySnapshot = new UsageMetricsDailySnapshot(id);
+    usageMetricsDailySnapshot.dailyActiveUsers = 0;
+    usageMetricsDailySnapshot.cumulativeUniqueUsers =
+      protocol.cumulativeUniqueUsers;
+    usageMetricsDailySnapshot.cumulativeUniqueDepositors =
+      protocol.cumulativeUniqueDepositors;
+    usageMetricsDailySnapshot.cumulativeUniqueBorrowers =
```

Comment:
For all of the snapshots, the fields that mirror the market and protocol entities should be on the outside of the if block so it gets updated every time you call the function.

Otherwise I don't see these fields being accumulated for snapshots

Reply:
Have you pushed your latest changes? There are no changes in this file